### ISSUE TITLE
Container builds are failing after several hours

### ISSUE THREAD

**Issue by TravisBowers** [adamrehn/ue4-docker#200]
Output of the `ue4-docker info` command:

```
$ ue4-docker info
ue4-docker version:         0.0.86 (latest available version is 0.0.89)
Operating system:           Linux (Ubuntu 20.04.2 LTS, 5.4.0-84-generic)
Docker daemon version:      20.10.7
NVIDIA Docker supported:    Yes
Maximum image size:         No limit detected
Available disk space:       207.06 GiB
Total system memory:        15.26 GiB physical, 4 GiB virtual
Number of processors:       36 physical, 36 logical
```

Additional details:

- Are you accessing the network through a proxy server? **No**

I am seeing failures to build ue4 images with the following command: 
```
$ sudo ue4-docker build custom:4.25-pixelstreaming -repo=https://github.com/ImmortalEmperor/UnrealEngine.git -branch=4.25-pixelstreaming --cuda=10.2 --no-engine
```

These attempts fail after several hours with:

```
  Took 457.237486s to run UE4Editor, ExitCode=137
  SafeDeleteFile /home/ue4/Library/Logs/Unreal Engine/LocalBuildLogs/BuildDerivedDataCache/DerivedDataCache-2021.09.08-18.08.33.txt
  SafeCopyFile /home/ue4/UnrealEngine/Engine/Programs/AutomationTool/Saved/DerivedDataCache-2021.09.08-18.00.56.txt /home/ue4/Library/Logs/Unreal Engine/LocalBuildLogs/BuildDerivedDataCache/DerivedDataCache-2021.09.08-18.08.33.txt
  SafeDeleteFile /home/ue4/UnrealEngine/Engine/Programs/AutomationTool/Saved/DerivedDataCache-2021.09.08-18.00.56.txt
  Editor terminated with exit code 137 (signal 9) while running DerivedDataCache; see log /home/ue4/Library/Logs/Unreal Engine/LocalBuildLogs/BuildDerivedDataCache/DerivedDataCache-2021.09.08-18.08.33.txt
  CommandletException: Editor terminated with exit code 137 (signal 9) while running DerivedDataCache; see log /home/ue4/Library/Logs/Unreal Engine/LocalBuildLogs/BuildDerivedDataCache/DerivedDataCache-2021.09.08-18.08.33.txt
    at AutomationTool.CommandUtils.RunCommandlet (Tools.DotNETCommon.FileReference ProjectName, System.String UE4Exe, System.String Commandlet, System.String Parameters, System.String& DestLogFile) [0x0076d] in <40220906006c4f6fa1f3b90411bc730c>:0
    at AutomationTool.CommandUtils.RunCommandlet (Tools.DotNETCommon.FileReference ProjectName, System.String UE4Exe, System.String Commandlet, System.String Parameters) [0x00001] in <40220906006c4f6fa1f3b90411bc730c>:0
    at AutomationTool.CommandUtils.DDCCommandlet (Tools.DotNETCommon.FileReference ProjectName, System.String UE4Exe, System.String[] Maps, System.String TargetPlatform, System.String Parameters) [0x00046] in <40220906006c4f6fa1f3b90411bc730c>:0
    at BuildDerivedDataCache.ExecuteBuild () [0x0034d] in <24122249d9234a06bb42e54b2ecafcfb>:0
    at AutomationTool.BuildCommand.Execute () [0x00001] in <40220906006c4f6fa1f3b90411bc730c>:0
    at AutomationTool.Automation.Execute (System.Collections.Generic.List`1[T] CommandsToExecute, System.Collections.Generic.Dictionary`2[TKey,TValue] Commands) [0x0007c] in <40220906006c4f6fa1f3b90411bc730c>:0
    at AutomationTool.Automation.Process (System.String[] Arguments, UnrealBuildTool.StartupTraceListener StartupListener) [0x00286] in <40220906006c4f6fa1f3b90411bc730c>:0
    at AutomationTool.Program.MainProc (System.String[] Arguments, UnrealBuildTool.StartupTraceListener StartupListener) [0x00001] in <39e89c519ff8445996d73015a03861d8>:0
    at AutomationTool.Program+<>c__DisplayClass1_0.<Main>b__2 () [0x00000] in <39e89c519ff8445996d73015a03861d8>:0
    at AutomationTool.InternalUtils.RunSingleInstance (System.Func`1[TResult] Main) [0x000a0] in <40220906006c4f6fa1f3b90411bc730c>:0
    at AutomationTool.Program.Main (System.String[] Arguments) [0x0020a] in <39e89c519ff8445996d73015a03861d8>:0
  AutomationTool exiting with ExitCode=1 (Error_Unknown)
Took 8870.550059s to run mono, ExitCode=1
AutomationTool exiting with ExitCode=1 (Error_Unknown)
RunUAT ERROR: AutomationTool was unable to run successfully.
The command '/bin/sh -c ./Engine/Build/BatchFiles/RunUAT.sh BuildGraph -target="Make Installed Build Linux" -script=Engine/Build/InstalledEngineBuild.xml -set:HostPlatformOnly=true -set:WithDDC=$BUILD_DDC  &&  rm -R -f /home/ue4/UnrealEngine/LocalBuilds/InstalledDDC && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''' returned a non-zero code: 1
[ue4-docker build] Error: failed to build image "adamrehn/ue4-minimal:4.25-pixelstreaming-cudagl10.2".
```
Any suggestion as to why this would be failing would be greatly appreciated.

**Comment by slonopotamus** [comment]
> signal 9

This means the process was killed externally by something. 99% you ran out of memory and OOM killer killed he process. The evidence should be visible `dmesg`.

**Comment by TravisBowers** [comment]
Thank you. I've increased my memory to core ratio and disk space. I am now able to build minimal images. However, I'm now noticing that I get an entirely different failure mode when I attempt to build full images. 

```
Building package "MediaIPC-ue4/0.0.2@adamrehn/4.25"...
['conan', 'install', 'MediaIPC-ue4/0.0.2@adamrehn/4.25', '--profile=ue4.25-Linux-x86_64-unknown-linux-gnu', '--build=missing']
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=clang
compiler.libcxx=libc++
compiler.version=9
os=Linux
os_build=Linux
[options]
[build_requires]
*: toolchain-wrapper/ue4@adamrehn/4.25
[env]
UNREAL_ENGINE_VERSION=4.25
boost/1.67.0@conan/stable: Not found in local cache, looking in remotes...
boost/1.67.0@conan/stable: Trying with 'conancenter'...
ERROR: Failed requirement 'boost/1.67.0@conan/stable' from 'MediaIPC-ue4/0.0.2@adamrehn/4.25'
ERROR: Unable to find 'boost/1.67.0@conan/stable' in remotes
Traceback (most recent call last):
  File "/usr/local/bin/ue4", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/dist-packages/ue4cli/cli.py", line 222, in main
    SUPPORTED_COMMANDS[command]['action'](manager, args)
  File "/usr/local/lib/python3.6/dist-packages/conan_ue4cli/main.py", line 48, in main
    SUBCOMMANDS[subcommand]['function'](manager, args[1:])
  File "/usr/local/lib/python3.6/dist-packages/conan_ue4cli/commands/build.py", line 130, in build
    builder.build(name, version, args.options if args.options is not None else [])
  File "/usr/local/lib/python3.6/dist-packages/conan_ue4cli/common/PackageBuilder.py", line 45, in build
    self._executor.execute(command, cwd=tempDir, check=True)
  File "/usr/local/lib/python3.6/dist-packages/conan_ue4cli/common/CommandExecutor.py", line 19, in execute
    return subprocess.run(command, **kwargs).returncode == 0
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['conan', 'install', 'MediaIPC-ue4/0.0.2@adamrehn/4.25', '--profile=ue4.25-Linux-x86_64-unknown-linux-gnu', '--build=missing']' returned non-zero exit status 1.
[ue4-docker build] [2021-09-09 17:39:36] [Available disk: 365.35 GiB] [Available memory: 11.31 GiB physical, 3.89 GiB virtual] [CPU usage: 5.90%]
The command '/bin/sh -c ue4 conan build MediaIPC-ue4 && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''' returned a non-zero code: 1
[ue4-docker build] Error: failed to build image "adamrehn/ue4-full:4.25-pixelstreaming-cudagl10.2".
```

Updated `ue4-docker info`:
```
$ ue4-docker info
ue4-docker version:         0.0.86 (latest available version is 0.0.90)
Operating system:           Linux (Ubuntu 20.04.2 LTS, 5.4.0-84-generic)
Docker daemon version:      20.10.7
NVIDIA Docker supported:    Yes
Maximum image size:         No limit detected
Available disk space:       365.35 GiB
Total system memory:        38.27 GiB physical, 4 GiB virtual
Number of processors:       18 physical, 18 logical
```

**Comment by jonpas** [comment]
You're now hitting #201, became a problem literally today.

**Comment by slonopotamus** [comment]
> I've increased my memory to core ratio and disk space. I am now able to build minimal images.

Nice, closing this issue then. #201 is a different story and most likely is going to be fixed by #202.
